model_server/app/cli.py

```diff
@@ -105,12 +133,21 @@ def stop_server():
         process = psutil.Process(pid)
 
         # Gracefully terminate the process
-        process.terminate()  # Sends SIGTERM by default
-        process.wait(timeout=10)  # Wait for up to 10 seconds for the process to exit
+        for child in process.children(recursive=True):
+            child.terminate()
+        process.terminate()
+
+        process.wait(timeout=20)  # Wait for up to 20 seconds for the process to exit
+
+        if process.is_running():
+            log.info(f"Process with PID {pid} is still running. Forcing shutdown.")
+            process.kill()  # Forcefully kill the process
 
         log.info(f"Model server with PID {pid} stopped.")
         os.remove(PID_FILE)
 
+        kill_process_on_port(port)
```

Comment:
curious, when will this be needed? If the PID file is not found? Then shouldn't we encapsulate that in the else block?

Reply:
PID file can be corrupted if you kept overwritting it. Happened to me and Adil so this function will be the last cleanse

Reply:
you are right too, im removing all pid now